openmc/material.py

```diff
@@ -41,11 +41,19 @@ class Material(object):
     name : str, optional
         Name of the material. If not specified, the name will be the empty
         string.
+    temperature : str, optional
+        The temperature identifier applied to this material. The units are
+        in Kelvin and the temperature rounded to the nearest integer.
+        For example, a tempreature of 293.6K would be provided as '294K'
 
     Attributes
     ----------
     id : int
         Unique identifier for the material
+    temperature : str
```

Comment:
I think that's a great strategy.  How does this interplay with the WMP?
WMP simply over-rides this as needed?

On Mon, Aug 29, 2016 at 2:09 PM, Paul Romano <notifications@github.com>
wrote:

> In openmc/material.py
> <https://github.com/mit-crpg/openmc/pull/712#discussion_r76656428>:
>
> >
> >      Attributes
> >      ----------
> >      id : int
> >          Unique identifier for the material
> > +    temperature : str
>
> How's this for user input:
>
> settings = openmc.Settings()
> settings.temperature = {
>   'default': 293.6,
>   'method': 'nearest',
>   'tolerance': 5.0
> }
>
> which would indicate that in the absence of material/cell temperatures, a
> temperature of 293.6 K is to be applied and cross sections within (293.6
> +/- 5) K would be acceptable (the nearest one chosen). The 'method' could
> also be set to 'interpolation' which would indicate that interpolation is
> to be applied. On the XML side, this would translate to:
>
> <temperature_default>293.6</temperature_default>
> <temperature_method>nearest</temperature_method>
> <temperature_tolerance>5.0</temperature_tolerance>
>
> Note that this is a marked departure from what we have now where the
> default is effectively specified in materials.xml. I just think it makes
> more sense to group it with the settings for method/tolerance.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/712/files/ced87d82560197af61f21ba02bb64ac1f02a51e3#r76656428>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA_TM8Y8IGrmG_sryw__gNskusnl-LHbks5qkyBjgaJpZM4Jr0jy>
> .
>